tests/functional/oauth_sign_in.js

```diff
@@ -344,7 +344,41 @@ define([
           assert.isTrue(/123done/i.test(text));
         })
         .end();
-    }
+    },
+
+    'oauth endpoint chooses the right auth flows': function () {
+      var self = this;
+
+      return self.get('remote')
+        .get(require.toUrl(OAUTH_APP))
+        .setFindTimeout(intern.config.pageLoadTimeout)
+
+        // use the 'Choose my sign-in flow for me' button
+        .findByCssSelector('#splash .sign-choose')
+        .click()
```

Comment:
I didn't realize you landed this new button on 123done! It'd be weird to see that option on a real site but it sure is useful for testing.